Scripts/extras/package_downloader.py

```diff
@@ -0,0 +1,134 @@
+
+class PackageDownloader(): 
+    @staticmethod
+    def DownloadAndUnpackPackage(package_name, package_hash, folder_target):
+        ''' given a public server URL list (semicolon-seperated), 
+            finds the package, if possible, and downloads and unpacks it to a given folder.
+            Note that this essentially mimics the server urls protocol used by cmake on the client
+            side and thus will also check buckets if s3:// urls are given
+            and will stop at the first one it finds.
+
+            Assumes that LY_PACKAGE_SERVER_URLS is set in the environment
+            Assumes that, if necessary, LY_AWS_PROFILE is set in the environment (if using s3 buckets)
+
+            returns True if it succeeded, False otherwise.
+         '''
+
+        def ComputeHashOfFile(file_path):
+            file_path = os.path.normpath(file_path)
+            hasher = hashlib.sha256()
+            hash_result = None
+            
+            # we don't follow symlinks here, this is strictly to check actual packages.
+            with open(file_path, 'rb') as afile:
+                buf = afile.read()
+                hasher.update(buf)
+                hash_result = hasher.hexdigest()
+        
+            return hash_result
+
+        # make sure a package with that name is not already present:
+        server_urls = os.environ.get("LY_PACKAGE_SERVER_URLS", default = "")
+
+        if not server_urls:
+            print(f"Server url list is empty - please set LY_PACKAGE_SERVER_URLS env var to semicolon-seperated list of urls to check")
+            print(f"Using default URL for convenience: https://d2c171ws20a1rv.cloudfront.net")
+            server_urls = "https://d2c171ws20a1rv.cloudfront.net"
+
+        download_location = pathlib.Path(folder_target)
+        package_file_name = package_name + ".tar.xz"
+        package_download_name = download_location / package_file_name
+        package_unpack_folder = download_location / package_name
+        
+        server_list = server_urls.split(';')
+
+        if os.path.exists(str(package_download_name)):
+            os.path.os.remove(str(package_download_name))
+      
+        if not os.path.exists(str(download_location)):
+            os.makedirs(str(download_location))
+
+        print(f"Downloading package {package_name}...")
+
+        for package_server in server_list:
+            if not package_server:
+                continue
+            full_package_url = package_server + "/" + package_file_name
+            print(f"    - attempting '{full_package_url}' ...")
+
+            try:
+                if package_server.startswith("s3://"):
+                    # we don't want to use temp buckets for package dependencies
+                    # you should depend on production (CDN) resources that have been verified!
+                    print(f"        - Note:  Ignoring {package_server} - S3 buckets should not be used for package dependencies.")
+                    continue
```

Comment:
This is on the user, not on the software to detect. There are legitimate long-term S3 stores that could be used for a dependency system for consumers creating their own 3P packages.

Reply:
If it can't plug to the existing S3 package infra yet, good enough reason.